src/internal/RequestFlowBuilder.ts

```diff
@@ -0,0 +1,191 @@
+import { of } from 'ipfs-only-hash';
+import log from 'loglevel';
+import { AquaCallHandler } from './AquaHandler';
+import { DEFAULT_TTL, RequestFlow } from './RequestFlow';
+
+export const loadVariablesService = 'load';
+const loadVariablesFn = 'load_variable';
+export const loadRelayFn = 'load_relay';
+const xorHandleService = '__magic';
+const xorHandleFn = 'handle_xor';
+export const relayVariableName = 'init_peer_relay';
+
+const wrapWithXor = (script: string): string => {
+    return `
+    (xor
+        ${script}
+        (xor
+            (match ${relayVariableName} ""
+                (call %init_peer_id% ("${xorHandleService}" "${xorHandleFn}") [%last_error%])
+            )
+            (seq 
+                (call ${relayVariableName} ("op" "identity") [])
+                (call %init_peer_id% ("${xorHandleService}" "${xorHandleFn}") [%last_error%])
+            )
+        )
+    )`;
+};
+
+class ScriptBuilder {
+    private script: string;
+
+    raw(script: string): ScriptBuilder {
+        this.script = script;
+        return this;
+    }
+
+    build(): string {
+        return this.script;
+    }
+}
+
+const wrapWithVariableInjectionScript = (script: string, fields: string[]): string => {
+    fields.forEach((v) => {
+        script = `
+(seq
+    (call %init_peer_id% ("${loadVariablesService}" "${loadVariablesFn}") ["${v}"] ${v})
+    ${script}
+)`;
+    });
+
+    return script;
+};
+
+const wrapWithInjectRelayScript = (script: string): string => {
+    return `
+(seq
+    (seq 
+        (call %init_peer_id% ("${loadVariablesService}" "${loadRelayFn}") [] ${relayVariableName})
+        (call %init_peer_id% ("op" "identity") [%init_peer_id%] init_peer_id)
+    )
+    ${script}
+)`;
+};
+
+export class RequestFlowBuilder {
+    private ttl: number = DEFAULT_TTL;
+    private variables = new Map<string, any>();
+    private handlerConfigs: Array<(handler: AquaCallHandler) => void> = [];
+    private buildScript: (sb: ScriptBuilder) => void;
+    private onTimeout: () => void;
+    private onError: (error: any) => void;
+
+    build() {
+        if (!this.buildScript) {
+            throw new Error();
+        }
+
+        const b = new ScriptBuilder();
+        this.buildScript(b);
+        let script = b.build();
+        script = wrapWithVariableInjectionScript(script, Array.from(this.variables.keys()));
+        script = wrapWithXor(script);
+        script = wrapWithInjectRelayScript(script);
```

Comment:
swap xor and inject

Reply:
we need relay variable before xor. So that's the correct order